book/continuous_integration.md

```diff
@@ -0,0 +1,192 @@
+# Automating Code Quality Assurance
+
+## Introduction
+
+There are various tasks which can be automated to increase the quality of code and make development easier and less tedious. Automating the running of unit tests is especially important to ensuring trust in your pipeline or package, by ensuring that all unit tests pass before every merge.
+
+(continuous-integration)=
+## Continous Integration
+
+Continuous integration (CI) describes the practice of frequently committing changes to your code. [CI tools](CI-tools) support this working pattern by automating routine quality assurance tasks. This includes verifying that your code successfully builds or installs and that your [code tests](testing_code.md) run successfully.
+
+[CI is often linked](https://www.atlassian.com/continuous-delivery/principles/continuous-integration-vs-delivery-vs-deployment) to:
+
+* Continuous delivery - ensuring that your code is fit for use after each integration
+* Continuous deployment - automatically deploying working code into production
+
+(automate-tests)=
+### Automate tests to reduce risk of errors
+
+Tests should be run whenever you make changes to your project.
+This ensures that changes do not break the existing, intended functionality of your code.
+However, it is easy to forget to run your tests at regular intervals.
+You're already putting effort into `commit`ing your changes to a version control system regularly.
+
+And you're right to think "surely this could be automated too?"
+Continuous integration can be used to automate testing, amongst other quality assurance measures, and can be triggered when changes are made to your remote version control repository.
+These tools can be used to ensure that all changes to a project are tested.
+Additionally, it allows others, who are reviewing your code, to see the results of your tests.
+
+Automation of routine tasks in this way reduces the effort required to merge changes onto the existing code base. This supports frequent commiting and merging of changes. As such, conflicts between multiple contributions should be minimal and review of these changes is simpler. Additionally, the execution environment for CI is defined in a CI workflow configuration, which improves reproducibility when running tests.
+
+## Pre-Commit and git hooks
+
+An alternative to continuous integration, is using a Git hook.
```

Comment:
I don't see pre-commits as an alternative to CI. I thinks its more an extra step of checks for QA.

'Another method for quality assurance, is using Git hooks.'

Reply:
I agree they are good used in tandem, but you should be able to use pre-commit hooks to run tests (although I've never tried it myself)

Reply:
Are pre-commit hooks included within the bracket of ci? Like if ci is the broad term, pre-commit hooks and github workflows are examples of ci. In which case, the initial sentence could be 

'One method of automating tests is the use of pre-commit hooks' 

The idea is this follows on from the automated test section before. If so, the automated test portion above should be generalised as it is conflating ci and workflows.

Reply:
pair coded with @joshlynchONS